### PR TITLE
Add QIT integration for running security tests locally

### PR DESCRIPTION
**Pull request by deepakpathania**
Fixes 155-gh-Automattic/woocommerce-payments-meta

#### Changes proposed in this Pull Request
- Introduce a script to run QIT security tests locally

#### Testing instructions

* Follow the `Setup and running` section in the newly added README.
* **Note** - QIT toolkit has a bug with accepting the zip files that we create in the build release so while that is fixed(hence the draft status), follow these steps:
  * Run the script normally and see the `Invalid zip` error.
  * Unzip the contents, delete the zip and then zip the unzipped folder again.
  * Remove the `npm run build:release` part from the `test:qit` script and run only with the shell script execution part.
* Ensure setup and testing instructions are clear enough. 
-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6486` or branch name `dev/qit-tests` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=dev/qit-tests) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 3d6205612fe0e010361d5a442bcc8e976de5764f
- Build time: 2023-06-13 07:09:16 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by deepakpathania** [comment]
Draft state till 453-gh-Automattic/compatibility-dashboard gets merged for the zip issue to be resolved.

File: changelog/dev-qit-tests [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: dev
+
+Add script to run QIT security tests locally.
```
Comment (jbordonado) [inline changelog/dev-qit-tests:4]: We don't need to include it in the next release's changelog, do we?

File: tests/qit/security.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+
+#Load local env variables if present.
+if [[ -f "$QIT_ROOT/config/local.env" ]]; then
+	. "$QIT_ROOT/config/local.env"
+fi
+
+# Check if QIT_USER and QIT_APP_PASSWORD are set and not empty
+if [[ -z $QIT_USER ]] || [[ -z $QIT_PASSWORD ]] ; then
+    echo "QIT_USER or QIT_APP_PASSWORD environment variables are not set or empty. Please set them before running the script."
```
Comment (jbordonado) [inline tests/qit/security.sh]: Maybe mention to "set them in a local env file" to improve the instruction a little?

**Review by jbordonado — APPROVED** [review]
It works well locally and the README is clear

I left two minor comments, but LGTM when the bug is solved on QIT's codebase

File: changelog/dev-qit-tests (hunk 1 above)
Comment (deepakpathania) [reply changelog/dev-qit-tests:4]: Mostly meant for release leads and internal use, so shouldn't be needed imo.

File: tests/qit/security.sh (hunk 2 above)
Comment (deepakpathania) [reply tests/qit/security.sh]: Done.

File: changelog/dev-qit-tests (hunk 1 above)
Comment (jbordonado) [reply changelog/dev-qit-tests:4]: I wasn't clear, but I meant it should maybe be added as an "insignificant" changelog, i.e. it should have a `Comment: ` before the text so it's not picked up for the release

**Pull request merged**